src/main/kotlin/lotto/domain/Rank.kt

```diff
@@ -0,0 +1,33 @@
+package lotto.domain
+
+enum class Rank(
+    val countOfMatch: Int,
+    val isMatchBonusNeed: Boolean,
+    val winningMoney: Int,
+) {
+    FIRST(6, false, 2_000_000_000),
+    SECOND(5, true, 30_000_000),
+    THIRD(5, false, 1_500_000),
+    FOURTH(4, false, 50_000),
+    FIFTH(3, false, 5_000),
+    MISS(0, false, 0), ;
+
+    companion object {
+        fun valueOf(
+            countOfMatch: Int,
+            matchBonus: Boolean,
+        ): Rank {
+            if (matchBonus) {
+                return entries.find { it.countOfMatch == countOfMatch } ?: MISS
+            }
+            return entries.filter { !it.isMatchBonusNeed }.find { it.countOfMatch == countOfMatch } ?: MISS
```

Comment:
이미 아실 것 같지만 다음과 같이 return 문을 밖으로 드러낼 수도 있습니다. 🙂 

```suggestion
            return if (matchBonus) {
                entries.find { it.countOfMatch == countOfMatch } ?: MISS
            } else {
                entries.filter { !it.isMatchBonusNeed }.find { it.countOfMatch == countOfMatch } ?: MISS
            }
```